packages/app/src/systems/Transaction/machines/transactionMachine.tsx

```diff
@@ -0,0 +1,223 @@
+  RECEIPTS_NOT_FOUND: 'Receipts not found for this transaction',
+};
+
+type GetTransactionResponse = {
+  txResponse: TransactionResponse;
+  tx: Transaction;
+  status?: TxStatus;
+  txId?: string;
+};
+
+type MachineContext = {
+  error?: string;
+  txResponse?: TransactionResponse;
+  txStatus?: TxStatus;
+  tx?: Transaction;
+  txResult?: TransactionResult<any>;
+  fee?: BN;
+  txId?: string;
+};
+
+type MachineServices = {
+  getTransaction: {
+    data: GetTransactionResponse;
+  };
+  getTransactionResult: {
+    data: TransactionResult<any>;
+  };
+};
+
+type MachineEvents = {
+  type: 'GET_TRANSACTION';
+  input: { providerUrl?: string; txId?: string };
+};
+
+export const transactionMachine = createMachine(
+  {
+    // eslint-disable-next-line @typescript-eslint/consistent-type-imports
+    tsTypes: {} as import('./transactionMachine.typegen').Typegen0,
+    schema: {
+      context: {} as MachineContext,
+      services: {} as MachineServices,
+      events: {} as MachineEvents,
+    },
+    predictableActionArguments: true,
+    id: '(machine)',
+    initial: 'idle',
+    states: {
+      idle: {
+        on: {
+          GET_TRANSACTION: [
+            {
+              cond: 'isInvalidTxId',
+              actions: 'assignInvalidTxIdError',
+            },
+            {
+              target: 'fetching',
+            },
+          ],
+        },
+      },
+      fetching: {
+        entry: 'clearError',
+        invoke: {
+          src: 'getTransaction',
+          data: (_, event: MachineEvents) => ({
+            input: event.input,
+          }),
+          onDone: [
+            {
+              actions: ['assignGetTransactionError'],
+              target: 'idle',
+              cond: FetchMachine.hasError,
+            },
+            {
+              actions: ['assignGetTransactionResponse'],
+              target: 'fetchingResult',
+            },
+          ],
+        },
+      },
+      fetchingResult: {
+        invoke: {
+          src: 'getTransactionResult',
+          data: (ctx) => ({
+            input: {
+              txResponse: ctx.txResponse,
+            },
+          }),
+          onDone: [
+            {
+              actions: ['assignGetTransactionResultError'],
+              target: 'idle',
+              cond: FetchMachine.hasError,
+            },
+            {
+              actions: ['assignGetTransactionResult'],
+              target: 'idle',
+            },
+          ],
+        },
+      },
+    },
+  },
+  {
+    actions: {
+      assignInvalidTxIdError: assign({
+        error: (_) => TRANSACTION_ERRORS.INVALID_ID,
+      }),
+      assignGetTransactionError: assign({
+        error: (_) => TRANSACTION_ERRORS.NOT_FOUND,
+      }),
+      assignGetTransactionResultError: assign({
+        error: (_) => TRANSACTION_ERRORS.RECEIPTS_NOT_FOUND,
+      }),
+      clearError: assign({
+        error: (_) => TRANSACTION_ERRORS.RECEIPTS_NOT_FOUND,
+      }),
```

Comment:
I think `clearError` should return undefined or null

Reply:
thanks! ctrlcctrlv feelings 😳